sentence_transformers/SentenceTransformer.py

```diff
@@ -1939,6 +1961,47 @@ def _keys_to_ignore_on_save(self) -> list[str]:
 
     def gradient_checkpointing_enable(self, gradient_checkpointing_kwargs=None) -> None:
         # Propagate the gradient checkpointing to the transformer model
-        for module in self:
-            if isinstance(module, Transformer):
-                return module.auto_model.gradient_checkpointing_enable(gradient_checkpointing_kwargs)
+        for child in self.modules():
+            if isinstance(child, PreTrainedModel):
+                return child.gradient_checkpointing_enable(gradient_checkpointing_kwargs)
+
+    def _get_model_type(
+        self,
+        model_name_or_path: str,
+        token: bool | str | None,
+        cache_folder: str | None,
+        revision: str | None = None,
+        local_files_only: bool = False,
+    ) -> str | None:
+        """
+        Retrieves the model_type from the config_sentence_transformers.json file.
+
+        This is used to determine the appropriate loading method:
+        - SentenceTransformer: These models should be loaded with _load_sbert_model when used with SentenceTransformer class
+        - SparseEncoder: These models should be loaded with _load_auto_model when used with SentenceTransformer class
+
+        When a model type doesn't match the class being used to load it, we switch loading methods
+        to ensure compatibility.
+
+        Args:
+            model_name_or_path (str): The name or path of the pre-trained model.
+            token (Optional[Union[bool, str]]): The token to use for the model.
+            cache_folder (Optional[str]): The folder to cache the model.
+            revision (Optional[str], optional): The revision of the model. Defaults to None.
+            local_files_only (bool, optional): Whether to use only local files. Defaults to False.
+
+        Returns:
+            Optional[str]: The model type (SentenceTransformer or SparseEncoder) if available, None otherwise.
+        """
+        config_sentence_transformers_json_path = load_file_path(
+            model_name_or_path,
+            "config_sentence_transformers.json",
+            token=token,
+            cache_folder=cache_folder,
+            revision=revision,
+            local_files_only=local_files_only,
+        )
+
```

Comment:
```suggestion

        if config_sentence_transformers_json_path is None:
            return "SentenceTransformer"

```

The `config_sentence_transformers.json` doesn't always exist. If it doesn't, but `modules.json` does exist, then we should probably assume `"SentenceTransformer"`.